### PR TITLE
Pandemics backend

### PR DESCRIPTION
**Pull request by pacyL2K19**
## The Aim of this PR

- Pandemy controller, module, and resources: Added relationship between pandemy cases, provinces and health zones
- Province controller, module, and resources: Allows now to create, update and delete provinces from the dashboard 
- Health zones controllers module, and resources 
- Relationship between all of these elements
#### Frontend SIde
- The UI of the pandemy page that contains 
- > Form to create and update 
- > List to display [image 1]
- The state features prepared 
* Need to add API consumption
* Need to review the get health zones to feature in the backend

> Images bellow
list provinces
health_zpne creation
delete prov
create province
ui

@merki230 :checkered_flag:

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -215,8 +222,36 @@
...
 
   Route::group(['prefix' => 'pandemics'], function () {
-    Route::get('top-confirmed', 'PandemicController@getHealthZoneTopConfirmed');
+    Route::get('top-confirmed', 'HealthZoneController@getHealthZoneTopConfirmed');
   });
 });
 
+// Pandemy group
+
+Route::group(['prefix' => 'pandemy', 'middleware' => 'auth:dashboard'], function () {
```
Comment (merki230) [inline routes/api.php:231]: Test bine que me middleware auth:dashboard marche bien

File: routes/api.php [hunk 2]
```diff
@@ -215,8 +222,36 @@
...
+  Route::get('/', 'PandemicController@index');
+  Route::get('/filter', 'PandemicController@filter');
+  // Route::get('/health_zones', 'PandemicController@health_zone');
+  Route::post('/', 'PandemicController@store');
+  Route::get('/{pandemy_id}', 'PandemicController@show');
+  Route::put('/{pandemy_id}', 'PandemicController@update');
+  Route::delete('/{pandemy_id}', 'PandemicController@destroy');
+  Route::get('/filter/health_zone', 'PandemicController@filter_by_health_zone');
```
Comment (merki230) [inline routes/api.php:239]: les routes statique doivent être toujours avant celle dynmique

File: routes/api.php [hunk 3]
```diff
@@ -215,8 +222,36 @@
...
+
+  // Health zone sub-zone
+
+  Route::group(['prefix' => 'health_zones'], function () {
+    Route::post('/', 'HealthZoneController@create');
+    Route::get('/', 'HealthZoneController@index');
+    Route::put('/{health_zone_id}', 'HealthZoneController@update');
+    Route::delete('/{health_zone_id}', 'HealthZoneController@destroy');
+    Route::get('/filter', 'HealthZoneController@filter');
```
Comment (merki230) [inline routes/api.php:248]: les routes statique doivent être toujours avant celle dynmiqu

**Review by merki230 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/HealthZoneController.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            return response()->json($pandemics, 200, [], JSON_NUMERIC_CHECK);
+        } catch (\Throwable $th) {
+            if (env('APP_DEBUG') == true) {
+                return response($th)->setStatusCode(500);
+            }
+            return response($th->getMessage())->setStatusCode(500);
+        }
+    }
+
+    public function health_zone(Request $request)
```
[the remark this answers is not in the document]
Comment (pacyL2K19) [reply app/Http/Controllers/HealthZoneController.php:52]: Elle est inutilisee, j'avais trouvE ca
Dans le PR qui suit j'ai corrigE ca

File: app/Http/Controllers/HealthZoneController.php [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                return response($th)->setStatusCode(500);
+            }
+            return response($th->getMessage())->setStatusCode(500);
+        }
+    }
+
+    public function index()
+    {
+        try {
+            $health_zones = HealthZone::orderBy('id', 'DESC')->paginate(15);
```
[the remark this answers is not in the document]
Comment (pacyL2K19) [reply app/Http/Controllers/HealthZoneController.php:68]: Noted

**Comment by pacyL2K19** [comment]
[This](https://github.com/DRC-COVID19/stopcoronavirus.cd/pull/164) PR contains everything fixed about this branch

**Pull request closed without merging**